bitcoin/src/blockdata/script/owned.rs

```diff
@@ -100,8 +100,8 @@ impl ScriptBuf {
     /// Pushes the slice without reserving
     fn push_slice_no_opt(&mut self, data: &PushBytes) {
         // Start with a PUSH opcode
-        match data.len() as u64 {
-            n if n < opcodes::Ordinary::OP_PUSHDATA1 as u64 => {
+        match u32::try_from(data.len()).expect("tried to put a 4bn+ sized object into a script!") {
```

Comment:
That `expect` is unreachable because of guarantees provided by `PushBytes`. It'd be better to have `len_u32()` method on `PushBytes` that does the cast and then call it from wherever this guarantee is used. Note that I'd prefer to use `as` there not `try_from`/`try_into` since the compiler doesn't know about our invariant.

Reply:
> That expect is unreachable because of guarantees provided by PushBytes

Could there be a test case that demonstrates this?

Reply:
`kani` proof should be feasible here.

Reply:
Oh we can just remove the catch all pattern all together. Done in #3126 because it is sufficiently different to this PR.

Reply:
Yes but expressing it via `u32` still makes sense.